spec/stepmod/change_spec.rb

```diff
@@ -0,0 +1,148 @@
+require "spec_helper"
+require "stepmod/utils/change"
+
+RSpec.describe Stepmod::Utils::Change do
+  describe "Instance methods" do
+    subject do
+      described_class.new(
+        stepmod_dir: stepmod_dir,
+        schema_name: "test_schema",
+        type: type,
+        resource: resource,
+      )
+    end
+
+    let(:stepmod_dir) { fixtures_path("stepmod_terms_mock_directory") }
+
+    describe "#resource?" do
+      let(:type) { "schema" }
+
+      context "when is a resource" do
+        let(:resource) { true }
+
+        it { expect(subject.resource?).to be(true) }
+      end
+
+      context "when is a module" do
+        let(:resource) { false }
+
+        it { expect(subject.resource?).to be(false) }
+      end
+    end
+
+    describe "#module?" do
+      let(:type) { "arm" }
+
+      context "when is a module" do
+        let(:resource) { false }
+
+        it { expect(subject.module?).to be(true) }
+      end
+
+      context "when is a resource" do
+        let(:resource) { true }
+
+        it { expect(subject.module?).to be(false) }
+      end
+    end
+
+    describe "#add_change_edition" do
+      let(:type) { "arm" }
+      let(:resource) { false }
+
+      let(:change_edition) do
+        {
+          version: "2",
+          description: "Test Description",
+          additions: [{ type: "ADD", name: "FooBar" }],
+          modifications: [{ type: "MODIFY", name: "FooBar" }],
+          deletions: [{ type: "DELETE", name: "FooBar" }],
+        }
+      end
+
+      it "should add change_edition to change" do
+        expect { subject.add_change_edition(change_edition) }
+          .to change { subject.change_editions.count }.from(0).to(1)
+      end
+    end
+
+    describe "#to_h" do
+      let(:type) { "arm" }
+      let(:resource) { false }
+
+      let(:change_edition) do
+        {
+          version: "2",
+          description: "Test Description",
+          additions: [{ type: "ADD", name: "FooBar" }],
+          modifications: [{ type: "MODIFY", name: "FooBar" }],
+          deletions: [{ type: "DELETE", name: "FooBar" }],
+        }
+      end
+
+      let(:expected_output) do
+        {
+          "schema" => "test_schema",
+          "change_edition" => [
+            {
+              "version" => "2",
+              "description" => "Test Description",
+              "additions" => [{ type: "ADD", name: "FooBar" }],
+              "modifications" => [{ type: "MODIFY", name: "FooBar" }],
+              "deletions" => [{ type: "DELETE", name: "FooBar" }],
+            },
+          ],
+        }
+      end
+
```

Comment:
Can you add specs that tests these cases described in #146?

## When a `<modified.object>` comes with `<description>`

```xml
      <schema.changes schema_name="geometry_schema">
         <schema.modifications>
            <modified.object type="ENTITY" name="curve_bounded_surface">
               <description>
WHERE 'WR4' expression changed</description>
            </modified.object>
            <modified.object type="ENTITY" name="locally_refined_spline_curve">
               <description>Add WHERE 'WR3'</description>
            </modified.object>
            <!-- ... -->
         </schema.modifications>
      </schema.changes>
```

## When a `<modified.object>` has `interfaced.items` attribute

```xml
      <schema.changes schema_name="topology_schema">
         <!-- ... -->
         <schema.deletions>
            <modified.object name="measure_schema"
                             type="REFERENCE_FROM"
                             interfaced.items="length_measure"/>
            <modified.object name="edge_with_length" type="ENTITY"/>
         </schema.deletions>
      </schema.changes>
```

## When a `<modified.change>` item only contains text

```xml
   <mapping.changes>
      <mapping.change>Annotation_placeholder_occurrence mapping was added;</mapping.change>
      <mapping.change>Dimension_curve_terminator_to_projection_curve_associativity mapping was modified;</mapping.change>
      <mapping.change>Draughting_model_item_association mapping was modified;</mapping.change>
      <mapping.change>Draughting_model_item_association_with_placeholder mapping was added;</mapping.change>
      <mapping.change>Geometric_placeholder_set mapping was added.</mapping.change>
   </mapping.changes>
```